File: Backend/backendwalud/app/Models/DoctorProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DoctorProfile extends Model
{
    protected $fillable = [
        'user_id',
        'rethus',
        'formacion_academica',
        'areas_enfoque',
        'horarios_atencion',
        'ubicaciones_consulta',
        'perfil_completo',
    ];

    protected $casts = [
        'formacion_academica'  => 'array',
        'areas_enfoque'        => 'array',
        'horarios_atencion'    => 'array',
        'ubicaciones_consulta' => 'array',
        'perfil_completo'      => 'boolean',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function checkIfComplete(): bool
    {
        return !empty($this->rethus) &&
               !empty($this->formacion_academica) &&
               !empty($this->areas_enfoque);
    }
}

File: Frontend/frontendwalud/lib/models/doctor_profile.dart
class DoctorProfile {
  final int?         id;
  final int?         userId;
  final String?      rethus;
  final List<Map<String, dynamic>> formacionAcademica;
  final List<String> areasEnfoque;
  final Map<String, dynamic>?      horariosAtencion;
  final List<Map<String, dynamic>> ubicacionesConsulta;
  final bool         perfilCompleto;
  final int          totalConsultas;
  final List<Map<String, dynamic>> ultimosPacientes;

  DoctorProfile({
    this.id,
    this.userId,
    this.rethus,
    this.formacionAcademica = const [],
    this.areasEnfoque       = const [],
    this.horariosAtencion,
    this.ubicacionesConsulta = const [],
    this.perfilCompleto      = false,
    this.totalConsultas      = 0,
    this.ultimosPacientes    = const [],
  });

  factory DoctorProfile.fromJson(Map<String, dynamic> json) {
    return DoctorProfile(
      id:                  json['id'],
      userId:              json['user_id'],
      rethus:              json['rethus'],
      formacionAcademica:  List<Map<String, dynamic>>.from(json['formacion_academica'] ?? []),
      areasEnfoque:        List<String>.from(json['areas_enfoque'] ?? []),
      horariosAtencion:    json['horarios_atencion'] != null
                               ? Map<String, dynamic>.from(json['horarios_atencion'])
                               : null,
      ubicacionesConsulta: List<Map<String, dynamic>>.from(json['ubicaciones_consulta'] ?? []),
      perfilCompleto:      json['perfil_completo'] == true,
      totalConsultas:      json['total_consultas'] ?? 0,
      ultimosPacientes:    List<Map<String, dynamic>>.from(json['ultimos_pacientes'] ?? []),
    );
  }

  Map<String, dynamic> toJson() => {
    'rethus':               rethus,
    'formacion_academica':  formacionAcademica,
    'areas_enfoque':        areasEnfoque,
    'horarios_atencion':    horariosAtencion,
    'ubicaciones_consulta': ubicacionesConsulta,
  };
}